cometd-java/cometd-java-server/cometd-java-server-http/cometd-java-server-http-jakarta/src/main/java/org/cometd/server/http/jakarta/JakartaCometDResponse.java

```diff
@@ -85,6 +86,29 @@ public void onError(Throwable failure) {
 
         @Override
         public void write(boolean last, byte[] bytes, Promise<Void> promise) {
+            // The first write is initiated outside of onWritePossible(),
+            // and that may cause problems, so it is delayed until
+            // onWritePossible() is called for the first time, which
+            // happens because of the call to setWriteListener().
```

Comment:
I think you should add to the comment that the first write cannot race with `onWritePossible()` nor `onError()` because `outputStream.isReady()` hasn't been called yet.